compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -17623,6 +17623,14 @@ private STNode parseAsListBindingPattern(STNode openBracket, List<STNode> member
 
     private STNode parseAsListBindingPattern(STNode openBracket, List<STNode> memberList) {
         memberList = getBindingPatternsList(memberList);
+        STNode firstMember = memberList.get(0);
+        if (firstMember.kind == SyntaxKind.FIELD_BINDING_PATTERN) {
```

Comment:
It is confusing we are doing this check only on the first member where we have a list. We need to do this for all the members. Also, we seem to have several places where we call ` getBindingPatternsList()` and create a list-binding-pattern on top of that. We need to handle this in a single place.

Reply:
Maybe we can pass a flag `isListBP` to  `getBindingPatternsList()` and `getBindingPattern()` and execute the default case if the node is not compatible member of `list-binding-pattern`.

Reply:
Updated